lib/sequent/core/helpers/association_validator.rb

```diff
@@ -26,7 +26,16 @@ def validate(record)
             if value && !value.kind_of?(Array) && record.respond_to?(:attributes) && !value.kind_of?(record.attributes[association])
               record.errors[association] = "is not of type #{record.attributes[association]}"
             elsif value && value.kind_of?(Array)
-              record.errors[association] = "is invalid" if value.any? { |v| not v.valid? }
+              item_type = record.class.type_for(association).item_type
+              record.errors[association] = "is invalid" if value.any? do |v|
```

Comment:
this inline `if` seems less handy now that it's got a multiline block on the end of it